### PR TITLE
wifi: Flash size optimizations

### PR DESCRIPTION
**Pull request by krish2718**
Use separate patch file for System + Raw modes

### CODE REVIEW

File: drivers/wifi/nrf700x/Kconfig [hunk 1]
```diff
@@ -31,8 +31,9 @@ config WIFI_NRF700X_BUS_LOG_LEVEL
 
 choice NRF700X_OPER_MODES
 	bool "nRF700x operating modes"
-	default NRF700X_SYSTEM_MODE if WPA_SUPP
+	default NRF700X_SYSTEM_MODE if WPA_SUPP && !(NRF700X_RAW_DATA_TX || NRF700X_RAW_DATA_RX || NRF700X_PROMISC_DATA_RX)
```
[the remark this answers is not in the document]
Comment (krish2718) [reply drivers/wifi/nrf700x/Kconfig:34]: These choices map 1-1 to the FW patches variants, so, not much room, but will try and see if I can make it simpler.

[the remark this answers is not in the document]
Comment (ioannisg) [reply drivers/wifi/nrf700x/Kconfig:34]: Here is my suggestion @krish2718 regarding the structure
- Operating modes
  - SCAN
  - SCAN+RAW 
  - SCAN+STA
  - SCAN+STA+RAW
  - SCAN+STA+SoftAP
  - SCAN+STA+SoftAP+RAW 

Clearly we do not need to have, _today_,  a separate patch variant for each mode but we could have the Kconfig structure ready. You could implement this structuring and exercise it for the "distinction" between SCAN+STA and SCAN+STA+RAW :)

[the remark this answers is not in the document]
Comment (krish2718) [reply drivers/wifi/nrf700x/Kconfig:34]: We don't want to maitain too many combinations as it adds to the maintenance and also needs to be discussed internally with the FW team, so, let's move to Jira to further discussions. The main intention of this PR is to remove RAW modes from default patch i.e., system mode.

**Review by rado17 — APPROVED** [review]

**Review by sachinthegreen — APPROVED** [review]

File: drivers/wifi/nrf700x/Kconfig [hunk 2]
```diff
@@ -52,9 +53,15 @@ config NRF700X_SCAN_ONLY
 config NRF700X_RADIO_TEST
 	bool "Radio test mode of the nRF700x driver"
 
+config NRF700X_SYSTEM_WITH_RAW_MODES
+	bool "Enable nRF700X system mode with raw modes"
```
Comment (ioannisg) [inline drivers/wifi/nrf700x/Kconfig:57]: @krish2718 don't you need to select NET_L2_WIFI_MGMT here?

File: drivers/wifi/nrf700x/Kconfig [hunk 3]
```diff
@@ -69,27 +76,23 @@ config NRF700X_AP_MODE
...
 	bool "Enable RAW RX sniffer operation in the driver"
 	select EXPERIMENTAL
-	depends on NRF700X_SYSTEM_MODE
 
 config NRF700X_PROMISC_DATA_RX
 	bool "Enable promiscuous RX sniffer operation in the driver"
 	select EXPERIMENTAL
-	depends on NRF700X_SYSTEM_MODE
+
+config NRF700X_DATA_TX
```
Comment (ioannisg) [inline drivers/wifi/nrf700x/Kconfig:93]: Any reason this is moved outside the `if NRF700X_SYSTEM_MODE || NRF700X_SYSTEM_WITH_RAW_MODES`

**Review by ioannisg — COMMENTED** [review]
I am OK with this patch to go in - it addresses a real problem.
That said, I would still like the Kconfig structure to be re-written, and become more general, so we can allow to extend easier in the future the number of patch variants. I can be done next time we add a new patch variant.

**Pull request merged**